### PR TITLE
Add ability to close preferences modal w/ esc key

### PR DESCRIPTION
**Pull request by dengeist**
Hey, New XKit!

Since the preferences window is a modal, I think users expect to be able to dismiss it with the keyboard. This PR is for just that! Thanks for working on this extension!

### CODE REVIEW

File: Extensions/xkit_preferences.js [hunk 1]
```diff
@@ -614,6 +614,11 @@ XKit.extensions.xkit_preferences = new Object({
 		$("#xkit-control-panel-shadow").click(function() {
 			XKit.extensions.xkit_preferences.close();
 		});
+		$(document).keydown(function(e) {
```
Comment (hobinjk) [inline Extensions/xkit_preferences.js]: This event listener should be attached in the `run` function (line 18). The `open` function potentially gets called several times per page which will cause this to unnecessarily duplicate event listeners.

It also should check for the existence of the `#xkit-control-panel` element before calling `close()` as `running`  is true when the `xkit_preferences` dialog isn't open.

An alternative fix would be to remove this event listener in the `close` function

Comment (dengeist) [inline Extensions/xkit_preferences.js]: Totally missed the way `#xkit-control-panel` affects run. Thanks, I'll get on these.

**Comment by dengeist** [comment]
Here you go @hobinjk! The `.close()` method detaches the keydown listener now.

File: Extensions/xkit_preferences.js [hunk 2]
```diff
@@ -734,6 +741,9 @@ XKit.extensions.xkit_preferences = new Object({
 			$("#xkit-control-panel").remove();
 		});
 
+		// Remove esc key listener attached on open
+		$(document).off('keydown');
```
Comment (hobinjk) [inline Extensions/xkit_preferences.js]: This removes every keydown listener attached to the document. Tumblr's code attaches a few keydown listeners (though I don't know what they use them for)

This should instead use the targeted version of `.off`, which requires making the keydown listener a property of xkit_preferences. Something like

```javascript
on_modal_keydown: function(event) {
// ....
},
```

would work properly and could be referenced in `on` and `off` as `XKit.extensions.xkit_preferences.on_modal_keydown`.

**Comment by dengeist** [comment]
Hey @hobinjk, sorry for letting this get stale. Followed your suggestion of adding an `on_modal_keydown` method to the `xkit_preferences` object. Let me know if this needs anything else.

File: Extensions/xkit_preferences.js [hunk 3]
```diff
@@ -172,6 +172,8 @@ XKit.extensions.xkit_preferences = new Object({
 
 		}
 
+
```
Comment (hobinjk) [inline Extensions/xkit_preferences.js]: nit: remove these blank lines

**Comment by dengeist** [comment]
Removed those lines and a couple others I found; thanks for pointing that out.

**Pull request merged**